Tests/FactoryTests/FactoryIsolationTests.swift

```diff
@@ -0,0 +1,45 @@
+import XCTest
+@testable import Factory
+
+private struct SomeSendableType: Sendable {}
+
+private struct AsyncInitWrapper<T>: Sendable {
+  let wrapped: @Sendable () async -> T
+}
+
+@MainActor
+private final class SomeMainActorType: Sendable {
+  init() {}
+}
+
+extension Container {
+  fileprivate var sendable: Factory<SomeSendableType> {
+    self { .init() }
+  }
+
+  @MainActor
+  fileprivate var mainActor: Factory<AsyncInitWrapper<SomeMainActorType>> {
+    self {
+      .init {
+        await SomeMainActorType()
+      }
+    }
+  }
+}
+
+final class FactoryIsolationTests: XCTestCase {
+
+  override func setUp() {
+    super.setUp()
+    Container.shared.reset()
+  }
+
+  func testInjectSendableDependency() {
+    let _: SomeSendableType = Container.shared.sendable()
+  }
+
+  func testInjectMainActorDependency() async {
+    let _: SomeMainActorType = await Container.shared.mainActor().wrapped()
```

Comment:
I took this example from https://github.com/hmlongco/Factory/commit/3ca72ff47173435b85e1df40b1dee8c34e6ab57e but this isn't viable... `Factory` would be caching `AsyncInitWrapper` but not the actual instance created by it.

Reply:
Unfortunately I can't think of a solution to this given that Swift (6) does not have a way to parameterize over an Actor.

Reply:
@NachoSoto have you had a look into [this macro](https://developer.apple.com/documentation/swift/isolation()) ?
Sorry, still trying to wrap my head around Swift Concurrency, so might be not related.